be/src/exec/pipeline/fragment_context.cpp

```diff
@@ -41,14 +41,20 @@ FragmentContext* FragmentContextManager::get_or_register(const TUniqueId& fragme
     }
 }
 
-void FragmentContextManager::register_ctx(const TUniqueId& fragment_id, FragmentContextPtr fragment_ctx) {
+Status FragmentContextManager::register_ctx(const TUniqueId& fragment_id, FragmentContextPtr fragment_ctx) {
     std::lock_guard<std::mutex> lock(_lock);
 
     if (_fragment_contexts.find(fragment_id) != _fragment_contexts.end()) {
-        return;
+        std::stringstream msg;
```

Comment:
Behavior is not the same as before, is there any problem?

Reply:
In previous code, the FragmentExecutor will continue to execute if the _fragment_contexts have been registered, this will lead to the same fragment  repeatedly executed for two times.
If the fragment context has been registerd before, we should abort the FragmentExecutor, the following prepare function should fail.
```
Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParams& common_request,
                                 const TExecPlanFragmentParams& unique_request) {
    DCHECK(common_request.__isset.desc_tbl);
    DCHECK(common_request.__isset.fragment);

    UnifiedExecPlanFragmentParams request(common_request, unique_request);

    bool prepare_success = false;
    DeferOp defer([this, &prepare_success]() {
        if (!prepare_success) {
            _fail_cleanup();
        }
    });
    RETURN_IF_ERROR(exec_env->query_pool_mem_tracker()->check_mem_limit("Start execute plan fragment."));

    RETURN_IF_ERROR(_prepare_query_ctx(exec_env, request));
    RETURN_IF_ERROR(_prepare_fragment_ctx(request));
    RETURN_IF_ERROR(_prepare_workgroup(request));
    RETURN_IF_ERROR(_prepare_runtime_state(exec_env, request));
    RETURN_IF_ERROR(_prepare_exec_plan(exec_env, request));
    RETURN_IF_ERROR(_prepare_global_dict(request));
    RETURN_IF_ERROR(_prepare_pipeline_driver(exec_env, request));

    RETURN_IF_ERROR(_query_ctx->fragment_mgr()->register_ctx(request.fragment_instance_id(), _fragment_ctx));
    prepare_success = true;

    return Status::OK();
}
```